sample/confidential_client_secret_sample.py

```diff
@@ -41,26 +42,36 @@
 
 config = json.load(open(sys.argv[1]))
 
-# Create a preferably long-lived app instance which maintains a token cache.
+persisted_cache = msal.SerializableTokenCache()  # See also https://msal-python.readthedocs.io/en/latest/#msal.SerializableTokenCache
```

Comment:
Moving the [conversation in #602](https://github.com/AzureAD/microsoft-authentication-library-for-python/issues/602#issuecomment-1745425229) to here.

> Agree on persisting the cache to disk for public client samples.
>
> But confidential clients don't really make that much use of I/O. Besides, in some Azure environments you don't even have access to the disk.

Do you mean a CCA sample shall NOT bother demonstrating the persist-to-disk pattern? I understand that it is not one-size-fits-all (which is also why MSAL Python's `token_cache` parameter allows caller to use different flavors of TokenCache), but I also think it is harmless to demonstrate a persist-to-disk pattern here. Besides, the current persist-to-disk sample does not actually incur I/O until the process exits and restarts.

The main reason to implement persist-to-disk pattern, however, is that [the dummy code snippet mentioned in #602](https://github.com/AzureAD/microsoft-authentication-library-for-python/issues/602#issuecomment-1744309904) (quoted below) feels silly:

```python
global_cache = msal.TokenCache()  # Now this becomes a static cache

# For whatever reason, multiple CCA will be created, but they can all share the same token cache
for i in range(10):
    cca = msal.ConfidentialClientApplication(..., cache=global_cache)
    token = cca.acquire_token_for_client(["scope"])
```

In most cases, app developer would better use this simpler approach:

```python
global_cca = msal.ConfidentialClientApplication(...)
# Now, global CCA's default in-memory token cache becomes a static cache,
# because the CCA is global/static, which is what we recommend.

for i in range(10):
    token = cca.acquire_token_for_client(["scope"])
```

So, perhaps these two action items?

* For most CCA samples, we shall simply change all `app = msal.ConfidentialClientApplication(...)` lines into `global_app = ...`, without using persisted cache.
* I will revisit the CCA sample using pre-computed client assertion, and see what changes are needed there. That is the (only?) scenario that it is not obvious how to persist CCA or its token cache.